pymc_marketing/mmm/delayed_saturated_mmm.py

```diff
@@ -814,3 +847,80 @@ def plot_channel_contributions_grid(
             ylabel="contribution",
         )
         return fig
+
+    def _validate_data(self, X, y=None):
+        return X
+
+    def sample_posterior_predictive(
+        self,
+        X_pred,
+        extend_idata,
+        combined,
+        include_last_observations: bool = False,
+        original_scale: bool = True,
+        **kwargs,
+    ):
+        """
+        Sample from the model's posterior predictive distribution.
+
+        Parameters
+        ---------
+        X_pred : array, shape (n_pred, n_features)
+            The input data used for prediction using prior distribution..
+        extend_idata : Boolean determining whether the predictions should be added to inference data object.
+            Defaults to False.
+        combined: Combine chain and draw dims into sample. Won't work if a dim named sample already exists.
+            Defaults to True.
+        include_last_observations: Boolean determining whether to include the last observations of the training data.
+            Defaults to False.
+        original_scale: Boolean determining whether to return the predictions in the original scale of the target variable.
+            Defaults to True.
+        **kwargs: Additional arguments to pass to pymc.sample_posterior_predictive
+
+        Returns
+        -------
+        posterior_predictive_samples : DataArray, shape (n_pred, samples)
+            Posterior predictive samples for each input X_pred
+        """
+        if include_last_observations:
+            X_pred = pd.concat(
+                [self.X.iloc[-self.adstock_max_lag :, :], X_pred], axis=0
+            )
+
+        self._data_setter(X_pred)
```

Comment:
I wonder if we should reset the data changes after sampling

Reply:
I'm hear this as setting the MutableData back to the X and y that it is fit on. i.e. self.X and self.y
Is that correct?
I think that would be good to preserve

Reply:
Added a `_reset_model_data` method to set to the original and it is called after the sample_posterior_predictive call

Reply:
On a second thought I see that all sampling methods need X as input, so I think we should revert this change. Otherwise we should reset in all the other methods. Just in this one doesn't make sense.

Apologies, I find the ModelBuilder API pretty confusing.